workload/pkg/info.go

```diff
@@ -224,6 +218,146 @@ func CreateResourceProviderFromAPI(ctx context.Context, kube kubernetes.Interfac
+			containers = append(containers, formatContainer(container, corev1.ContainerStatus{}, item.Spec.Template.GetCreationTimestamp()))
+		}
+		job := formatControllers("Job", item.Name, item.Namespace, string(item.UID), item.GetObjectMeta().GetOwnerReferences(), containers, item.Annotations, item.Labels)
+		interfaces = append(interfaces, job)
+	}
+
+	// Pods
+	pods, err := kube.CoreV1().Pods("").List(ctx, listOpts)
+	if err != nil {
+		logrus.Errorf("Error fetching Pods %v", err)
+		return nil, err
+	}
+
+	for _, item := range pods.Items {
+		var containers []ContainerResult
+		if item.Status.Phase != corev1.PodRunning && item.Status.Phase != corev1.PodPending {
+			continue
+		}
+		for _, container := range item.Spec.Containers {
+			for _, status := range item.Status.ContainerStatuses {
+				if status.Name == container.Name {
+					containers = append(containers, formatContainer(container, status, item.GetObjectMeta().GetCreationTimestamp()))
+					break
+				}
+			}
+		}
+		ownerReferences := item.GetObjectMeta().GetOwnerReferences()
+		ownerUID := getOwnerUID(ownerReferences)
+
+		if ownerUID != "" {
+			ownerController := controllerMap[ownerUID]
+			if ownerController != nil {
+				controllerMap[ownerUID].PodCount++
+				ownerContainers := controllerMap[ownerUID].Containers
+				if len(ownerContainers) == 0 || (len(containers) > 0 && containers[0].CreationTime.After(ownerContainers[0].CreationTime)) {
+					controllerMap[ownerUID].Containers = containers
+				}
+				continue
+			}
+			if ownerReferences[0].Kind == "Job" {
+				continue
+			}
+		}
+		pod := formatControllers("Pod", item.Name, item.Namespace, string(item.UID), ownerReferences, containers, item.Annotations, item.Labels)
+		controllerMap[pod.UID] = &pod
+		topController[pod.UID] = true
+	}
+
+	finalInterfaces := []ControllerResult{}
+	for id := range topController {
+		controller := controllerMap[id]
+		var count float64 = controller.PodCount
+		if controller.Kind != "CronJob" {
+			for _, childID := range children[id] {
+				count += controllerMap[childID].PodCount
+			}
+		} else {
+			// calculate min time
+			// Calculate max time
+			// Calculator avg runtime * count - 1
+			// Divide sum runtime by length of max-min
+			// If 0 jobs then 1 pod
+			// If 1 job then 1 pod
+			children := cronChildren[id]
+			if len(children) > 1 {
+				minTime := children[0].startTime
+				maxTime := children[0].startTime
+				var durationSum float64 = 0
+				for _, timeStamps := range cronChildren[id] {
+					durationSum += timeStamps.endTime.Sub(timeStamps.startTime).Seconds()
+					if timeStamps.startTime.Before(minTime) {
+						minTime = timeStamps.startTime
+					}
+					if timeStamps.startTime.After(maxTime) {
+						maxTime = timeStamps.startTime
+					}
+				}
+				totalTime := maxTime.Sub(minTime).Seconds()
+				durationSum = (durationSum / float64(len(children))) * float64(len(children)-1)
+				count = durationSum / totalTime
+			} else {
+				count = 1
```

Comment:
ditto on the `Job` calculation - maybe `duration / (now() - startTime)`? I think that's still an _over_ estimate in all cases...if the cj just ran, it'll give `1`, but if it only runs once/year, the pod count will slowly decay toward `0`.

Reply:
I like it. Yeah this feels like a good way to get even more accurate numbers.